### PR TITLE
Update monitor and scheduler to use cache setup in api

### PR DESCRIPTION
**Pull request by jmcshane**
This looks like were in good shape now, will discuss tomorrow with @benmoss about testing local runner

### CODE REVIEW

**Review by benmoss — CHANGES_REQUESTED** [review]

File: api/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package api
```
[the remark this answers is not in the document]
Comment (jmcshane) [reply api/kubernetes.go:1]: yeah, im going to create a `kubernetes` package

File: main.go [hunk 2]
```diff
@@ -16,11 +13,13 @@ import (
...
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/clientcmd"
+
+	restconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
 )
 
 var debug *bool = flag.Bool("debug", false, "debug logs")
-var maxInFlight *int = flag.Int("max-in-flight", 1, "max jobs in flight, 0 means no max")
+var maxInFlight *int32 = flag.Int32("max-in-flight", 1, "max jobs in flight, 0 means no max")
```
[the remark this answers is not in the document]
Comment (jmcshane) [reply main.go:22]: :+1:

File: main.go [hunk 3]
```diff
@@ -37,37 +36,25 @@ func main() {
...
-
-	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, nil)
-	clientConfig, err := kubeConfig.ClientConfig()
-	if err != nil {
-		log.Fatal("failed to create client config", zap.Error(err))
-	}
+	ctx := signals.SetupSignalHandler()
 
+	clientConfig := restconfig.GetConfigOrDie()
```
[the remark this answers is not in the document]
Comment (jmcshane) [reply main.go:41]: im confused...it says that it tries four files here including the HOME/.kube/config: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/config#GetConfigOrDie

File: api/kubernetes.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+func tagsToLabels(tags []string) []string {
+	labels := make([]string, len(tags))
+	for i, tag := range tags {
+		labels[i] = TagToLabel(tag)
+	}
+	return labels
+}
+
+func NewBuildkiteJobManagerOrDie(ctx context.Context, clientset kubernetes.Interface, tags ...string) *BuildkiteJobManager {
```
[the remark this answers is not in the document]
Comment (jmcshane) [reply api/kubernetes.go:42]: ill update this

**Comment by jmcshane** [comment]
@benmoss updated with your PR

**Review by benmoss — APPROVED** [review]

**Pull request merged**